SU2_CFD/src/numerics/CLookUpTable.cpp

```diff
@@ -0,0 +1,593 @@
+  vector<su2double> weights(2, 0);
+
+  vector<su2double> prog_enth_pairs(2 * n_points);
+
+  vector<unsigned long> next_triangle;
+
+  const vector<su2double>& prog = GetData(name_prog);
+  const vector<su2double>& enth = GetData(name_enth);
+
+  vector<unsigned long> result_ids;
+  vector<int> result_ranks;
+  vector<su2double> best_dist;
+
+  for (unsigned long i_point = 0; i_point < n_points; i_point++) {
+    points.push_back(i_point);
+
+    prog_enth_pairs.push_back(prog.at(i_point));
+    prog_enth_pairs.push_back(enth.at(i_point));
+  }
+
+  /* calculate weights for each triangle (basically a distance function) and
+   * build inverse interpolation matrices */
+  for (unsigned long i_triangle = 0; i_triangle < n_triangles; i_triangle++) {
+    next_triangle = triangles.at(i_triangle);
+
+    /* the query point is the weighted average of the vertexes of the triangle */
+    weights.at(0) = 0;
+    weights.at(1) = 0;
+
+    /* enthalpy */
+    weights.at(0) += enth.at(next_triangle.at(0));
+    weights.at(0) += enth.at(next_triangle.at(1));
+    weights.at(0) += enth.at(next_triangle.at(2));
+    weights.at(0) /= 3;
+
+    /* progress variable */
+    weights.at(1) += prog.at(next_triangle.at(0));
+    weights.at(1) += prog.at(next_triangle.at(1));
+    weights.at(1) += prog.at(next_triangle.at(2));
+    weights.at(1) /= 3;
+
+    interp_points.push_back(next_triangle);
+
+    // Now use the nearest 16 points to construct an interpolation function
+    // for each search pair option
+    vector<vector<su2double> > prog_interp_mat_inv(3, vector<su2double>(3, 0));
+    GetInterpMatInv(prog, enth, next_triangle, prog_interp_mat_inv);
+    interp_mat_inv_prog_enth.push_back(prog_interp_mat_inv);
+  }
+}
+
+void CLookUpTable::GetInterpMatInv(const vector<su2double>& vec_x, const vector<su2double>& vec_y,
+                                   vector<unsigned long>& point_ids, vector<vector<su2double> >& interp_mat_inv) {
+  vector<vector<su2double> > interp_mat(3, vector<su2double>(3, 0));
+
+  /* setup LHM matrix for the interpolation */
+  for (int i_point = 0; i_point < 3; i_point++) {
+    su2double x = vec_x.at(point_ids.at(i_point));
+    su2double y = vec_y.at(point_ids.at(i_point));
+
+    interp_mat.at(i_point).at(0) = 1;
+    interp_mat.at(i_point).at(1) = x;
+    interp_mat.at(i_point).at(2) = y;
+  }
+
+  /* invert the Interpolation matrix using Gaussian elimination with pivoting */
+  GaussianInverse(interp_mat, interp_mat_inv);
+
+  /* transpose the inverse */
+  su2double swap_helper;
+  for (int i = 0; i < 2; i++) {
+    for (int j = i + 1; j < 3; j++) {
+      swap_helper = interp_mat_inv.at(i).at(j);
+      interp_mat_inv.at(i).at(j) = interp_mat_inv.at(j).at(i);
+      interp_mat_inv.at(j).at(i) = swap_helper;
+    }
+  }
+}
+
+void CLookUpTable::GaussianInverse(vector<vector<su2double> >& mat, vector<vector<su2double> >& mat_inv) {
+  /* temp provides memory to invert mat */
+  vector<vector<su2double> > temp;
```

Comment:
Use the functions from blas_structure

Reply:
Creating a CSymmetricMatrix is OK for this? 
An then after the invert we can convert invM into an su2matrix using invM.StealData() ?

Reply:
That type is not compatible with su2double I think, it's hardcoded for passive double to call blas when available.
There is a function in blas_structure to invert a general matrix.
If the matrix is always 3*3 we should use a type of su2matrix with know size, there's an example of that in the NEMO numerics.

Reply:
I am now using CSquareMatrixCM in CLookupTable::GetInterpMatInv and temporarily convert it back to a <vector<vector>> at the end of the function to not break everything else.
I had to use SU2_TYPE::GetValue to put su2doubles into the matrix. Is this the 'hardcoded to passivedouble' that you mention?
The local Gaussian inverse routine has been removed.